static/js/search.js

```diff
@@ -63,7 +70,13 @@ exports.initialize = function () {
         matcher: function () {
             return true;
         },
-        updater: narrow_or_search_for_term,
+        updater: function (search_string) {
+            // Order is important here. narrow_or_search_for_term
+            // assumes that the pill for the selected typeahead has
+            // not been added.
```

Comment:
Can you explain why that's the case?  If I'm just reading this comment, I don't really know why `narrow_or_search_for_term` has that restriction, and maybe we can figure out a way so they don't depend on each other?  (I think the main thing is just to add to the comment.)

Reply:
narrow_or_search_for_term gets a search string from existing pills and obtains existing operators. Newly selected suggestion is added to those operators. If narrow_or_search_for_term was called after append_search_string, the existing search pills at the time for calling that function would also have the newly selected suggestion, and appending it again would cause duplication. 
I've also added above text to the comment.